src/main/java/frc/robot/commands/PlaceCone/toZero.java

```diff
@@ -0,0 +1,157 @@
+  private double[] values = {0, 0, 0};
+  private boolean angleDone = false;
+  private double p = .002;
+  private int noTarg = 0;
+  private double angleP = .03;
+  private double anglePDefault;
+  private GenericEntry anglePEntry;
+
+  private final ShuffleboardTab tab;
+
+  private ProfiledPIDController anglePid;
+
+  // second param on constraints is estimated, should be max accel, not max speed, but lets say it
+  // gets there in a second
+  private Constraints angleConstraints =
+      new Constraints(
+          DrivetrainConstants.MAX_ANGULAR_VELOCITY_RADIANS_PER_SECOND,
+          DrivetrainConstants.MAX_ANGULAR_VELOCITY_RADIANS_PER_SECOND);
+
+  //// second param on constraints is estimated, should be max accel, not max speed, but lets say it
+  // gets there in a second
+
+  private double angleI = 0;
+  private double angleD = 0;
+
+  /** Creates a new Rotate. */
+  public toZero(
+      DrivetrainSubsystem drivetrainSubsystem, PoseEstimatorSubsystem poseEstimatorSubsystem) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    this.poseEstimatorSubsystem = poseEstimatorSubsystem;
+    this.drivetrainSubsystem = drivetrainSubsystem;
+
+    anglePDefault = .03;
+
+    tab = Shuffleboard.getTab("Drive");
+
+    ShuffleboardLayout PIDConstants =
+        tab.getLayout("Rotate PID Constants", BuiltInLayouts.kList)
+            .withSize(2, 4)
+            .withPosition(8, 0);
+
+    if (PIDConstants.getComponents().isEmpty()) {
+
+      anglePEntry =
+          PIDConstants.add("Angle P Constant", anglePDefault)
+              .withWidget(BuiltInWidgets.kTextView)
+              .getEntry();
+
+    } else {
+      List<ShuffleboardComponent<?>> widgets = PIDConstants.getComponents();
+      anglePEntry = ((SimpleWidget) widgets.get(0)).getEntry();
+    }
+
+    addRequirements(poseEstimatorSubsystem, drivetrainSubsystem);
+    // addRequirements(drivetrainSubsystem);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    angleDone = false;
+
+    angleP = anglePEntry.getDouble(anglePDefault);
+
+    anglePid = new ProfiledPIDController(angleP, angleI, angleD, angleConstraints);
+
+    // configure rotation pid
+    System.out.println(poseEstimatorSubsystem.getAngle());
+    anglePid.reset(poseEstimatorSubsystem.getAngle());
+    anglePid.setGoal(180);
+    anglePid.setTolerance(.5);
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+
+    // gets drive values to align to 0 degrees, field oriented.
+    values = chassisValuesLower();
+    drivetrainSubsystem.drive(new ChassisSpeeds(values[0], values[1], values[2]));
+    // System.out.printf(
+    //     "X equals %.2f PID moves %.2f%n", poseEstimatorSubsystem.getAngle(), values[2]);
+    // setpoint and atgoal don't work, just brute forced.
+    if (Math.abs(180 - poseEstimatorSubsystem.getAngle()) < 3) {
```

Comment:
This will not work when the angle is near -179 - should terminate,  but won't.
Needs to mod by 360